protobuf-format.md

```diff
@@ -0,0 +1,251 @@
+# Protobuf Event Format for CloudEvents - Version 1.0
+
+## Abstract
+
+[Protocol Buffers][proto-home] is a mechanism for marshalling structured data.
+This document defines how CloudEvents are represented using [version 3][proto-3]
+of that specification.
+
+In this document the terms *Protocol Buffers*, *protobuf*, and *proto* are used
+interchangeably.
+
+## Status of this document
+
+This document is a working draft.
+
+## Table of Contents
+
+1. [Introduction](#1-introduction)
+2. [Attributes](#2-attributes)
+3. [Data](#3-data)
+4. [Transport](#4-transport)
+5. [Examples](#5-examples)
+
+## 1. Introduction
+
+[CloudEvents][ce] is a standardized and protocol-agnostic definition of the
+structure and metadata description of events. This specification defines how the
+elements defined in the CloudEvents specification are are represented using
+a protobuf schema. The  [schema][proto-schema] defines an envolope/body
+construct similar to that prescriped in the [JSON format][json-format].
+
+The [Attributes](#2-attributes) section describes the naming conventions and
+data type mappings for CloudEvents attributes for use as protobuf message
+properties.
+
+### 1.1. Conformance
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD",
+"SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be
+interpreted as described in [RFC2119][rfc2119].
+
+### 1.2 Content-Type
+
+There is no official IANA *media-type* designation for protobuf, as such this
+specification uses 'application/x-protobuf' to identify such content.
```

Comment:
There was a draft for protobuf media type here: https://tools.ietf.org/html/draft-rfernando-protocol-buffers-00.
We recently decided in our company to use the media type `application/protobuf` proposed in the draft above. 
Would you mind switching to that definition instead of the proposed `application/x-protobuf`?

```suggestion
specification uses 'application/protobuf' to identify such content based on the draft [Encoding rules and MIME type for Protocol Buffers](https://tools.ietf.org/html/draft-rfernando-protocol-buffers-00).
```

Reply:
Hi,

I certainly think this warrants further discussion, i've seen both application/protobuf and application/x-prototbuf used.

Looking at [Spring for example](https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/http/converter/protobuf/ProtobufHttpMessageConverter.html) they appear to use the x-prototbuf.

I'm not sure there's one correct answer to this :-(

Reply:
Sorry, my mistake - we mainly thought about the `datacontenttype` attribute.

I think the proposal made in the call today to use `application/cloudevents+protobuf` for the `Content-Type` http header field for structured mode is reasonable.

@JemDay : You can ignore my suggested change.